File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {

        int start = 0, end = nums.length - 1, count = 0;
        while (start <= end && end >= start) {
            if (nums[start] == val) {
                if (nums[start] == nums[end]) {
                    end--;
                } else {
                    int temp = nums[start];
                    nums[start++] = nums[end];
                    nums[end--] = temp;
                    count++;
                }
            } else {
                start++;
                count++;
            }
        }
        return count;
    }
}

File: 0027-remove-element/0027-remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
    let ans = [];
    nums.forEach((num) => {
        if (num != val) {
            ans.push(num);
        }
    })
    for (let i = 0; i < ans.length; i++) {
        nums[i] = ans[i];
    }
    return ans.length;
};